### ISSUE TITLE
R code generator uses factor levels for numeric subsetting

### ISSUE THREAD

**Issue by mwip** [ddionrails/tools#69]
#From the top of the script, reading data with `read.dta()` will create factor levels for variables containing letters. 
```R
pfad <- read.dta(file.path(path_in, "ppfad.dta"))
```

When subsetting the data based on such variables (e.g. $netto), R cannot use the numerical information within the `[xx]` directly. 
```R
pfad <- with(pfad, pfad[ (xnetto >= 10 & xnetto < 20) | (wnetto >= 10 & wnetto < 20) | (ynetto >= 10 & ynetto < 20) , ])
```
One possible solution might be to use the first characters of the factor levels to generate signed integers which could then be subsetted. 

```R
for (i in grep("netto", colnames(pfad))){
  pfad[,i] <- as.integer(gsub("\\[|\\]", "", substr(pfad[,i], 1, 4)))
}
```
However, it should be considered to copy the data bevorehand, in order to keep the information.

**Comment by kwenzig** [comment]
Zur Einordnung: Das ist übrigens ein bug, der den Skript-Generator unbrauchbar macht.

**Comment by kwenzig** [comment]
Will be solved, when mhebing/ddionrails2#370 is solved. Testversion is already working. Will be deployed soon. Thanks for this issue!
